### PR TITLE
Updated to Gradle 7

### PR DESCRIPTION
**Pull request by BomBardyGamer**
Does what it says in the title. I've also update shadow to go with this, and also switched every `compile` to `implementation` (since `compile` has been removed in 7), and switched to using `filter` to replace tokens in resource files, rather than the previously used `expand`.

### CODE REVIEW

**Comment by lucko** [comment]
Thanks for this - I will merge once I have a spare moment to ensure Jenkins is up-to-date and configure a different Gradle install

**Comment by lucko** [comment]
> switched to using filter to replace tokens in resource files, rather than the previously used expand.

Why was this, out of interest? @BomBardyGamer

**Comment by lucko** [comment]
Other than that, looks good, Jenkins is ready for Java 16 + Gradle 7 so I can merge now 😄

File: bukkit-legacy/build.gradle [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-    compile 'com.google.code.gson:gson:2.7'
-    compile 'com.google.guava:guava:19.0'
+    implementation project(':bukkit')
+    implementation 'com.google.code.gson:gson:2.7'
+    implementation 'com.google.guava:guava:19.0'
 }
 
 shadowJar {
-    archiveName = 'luckperms-bukkitlegacy.jarinjar'
+    archiveName 'luckperms-bukkitlegacy.jarinjar'
```
Comment (emilyy-dev) [inline bukkit-legacy/build.gradle:12]: Seeing that the gradle scripts are being reworked, this should be `archiveFileName` (same for the other cases); `get/setArchiveName` is deprecated

**Review by emilyy-dev — CHANGES_REQUESTED** [review]

**Comment by lucko** [comment]
Thanks :)

**Pull request merged**